repos/system_upgrade/common/actors/scancustommodifications/libraries/scancustommodifications.py

```diff
@@ -0,0 +1,154 @@
+import ast
+import os
+
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.common import dnfconfig
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import CustomModifications
+
+LEAPP_REPO_DIRS = ['/usr/share/leapp-repository']
+LEAPP_PACKAGES_TO_IGNORE = ['snactor']
+
+
+def _get_dirs_to_check(component):
+    if component == 'repository':
+        return LEAPP_REPO_DIRS
+    return []
+
+
+def _get_rpms_to_check(component=None):
+    # NOTE(ivasilev) Still not sure that library function is the way to go as looks like leapp file changes have
+    # to be handled differently, and configuration change has to be tracked only for leapp-repository as well.
+    # Let's keep it here for now but I am probably switching to LEAPP_PKG and LEAPP_REPO_PKG in the next commit.
+    leapp_installation = set(dnfconfig.get_leapp_packages()) - set(LEAPP_PACKAGES_TO_IGNORE)
+    leapp_repo = ['leapp-upgrade-el{source}toel{target}'.format(source=get_source_major_version(),
+                                                                target=get_target_major_version())]
+    if component == 'repository':
+        return leapp_repo
+    if component == 'framework':
+        return list(leapp_installation - set(leapp_repo))
+    return list(leapp_installation)
+
+
+def deduce_actor_name(a_file):
+    """
+    A helper to map an actor/library to the actor name
+    If a_file is an actor or an actor library, the name of the actor (name attribute of actor class) will be returned.
+    Empty string is returned if the file could not be associated with any actor.
+    """
+    if not os.path.exists(a_file):
+        return ''
+    # NOTE(ivasilev) Actors reside only in actor.py files, so AST processing any other file can be skipped.
+    # In case this function has been called on a non-actor file, let's go straight to recursive call on the assumed
+    # location of the actor file.
+    if a_file.endswith('actor.py'):
```

Comment:
@fernflower  Let's rather use `os.path.basename` to cover cornercases like `my_private_lib_actor.py` or something like that. Not sure what all names could be present e.g. in case of custom actors.

```suggestion
    if os.path.basename(a_file) == 'actor.py':
```

needs to `import os`